File: week1/pset1/cash.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void){

    // Declare user input
    float input;
    // Declare coin count
    int count = 0;
    // Check if user input is a
    // non-negative number
    do {
        input = get_float("Change owed:");
    }
    while(input < 0);
    // round input to the nearest cent
    int cents = round(input * 100);

    // Determine the minimum number of coins
    // that the change can be made
    do{
        if (cents >= 25){
            cents -= 25;
            count++;
        } else if (cents >= 10){
            cents -= 10;
            count++;
        } else if (cents >= 5){
            cents -= 5;
            count++;
        } else if (cents >= 1) {
            cents -= 1;
            count++;
        }
    } 
    while(cents != 0);
    printf("Number of coins: %i\n", count);

    
}

File: week6/pset6/cash/cash.py
from cs50 import get_float

change_input = get_float("Changed owed: ")

# Reprompt user if input is a non-negative number
while (change_input < 0):
    change_input = get_float("Changed owed: ")
    
cents = round(change_input * 100)
coins = 0
while (cents != 0):
    # Starting with the largest coin
    coins += cents // 25;
    cents = cents % 25;
    
    coins += cents // 10;
    cents = cents % 10;
    
    coins += cents // 5;
    cents = cents % 5;
    
    coins += cents // 1;
    cents = cents % 1;

print(coins)
    